src/lowering/desugar.ml

```diff
@@ -321,7 +321,7 @@ and call_system_func_opt name es obj_typ =
   List.find_map (fun es ->
     match es.it with
     | { S.vis = { it = S.System; _ };
-        S.dec = { it = S.LetD( { it = S.VarP id; note; _ }, _); at; _ };
+        S.dec = { it = S.LetD( { it = S.VarP id; note; _ }, _, None); at; _ };
```

Comment:
Are you sure once can't write

```
system let preupgrade = func () {} else { loop {}}
```
In which case this match won't apply as expected?

Reply:
Or is this ruled out by type error M0184?

Reply:
I removed the type error, but this is giving a syntax error (misplaced token "system")

Reply:
Fixed